docs/features/interceptors.md

```diff
@@ -66,29 +66,26 @@ Any "ordinary method" (i.e. with `MethodKind.Ordinary`) can have its calls inter
 
 File-local declarations of this type (`file class InterceptsLocationAttribute`) are valid and usages are recognized by the compiler when they are within the same file and compilation. A generator which needs to declare this attribute should use a file-local declaration to ensure it doesn't conflict with other generators that need to do the same thing.
 
-#### File paths
+In prior experimental releases of the feature, a well-known constructor signature `InterceptsLocation(string path, int line, int column)]` was also supported. Support for this constructor will be **dropped** prior to stable release of the feature.
 
-The *referenced syntax tree* of an `[InterceptsLocation]` is determined by normalizing the `filePath` argument value relative to the path of the containing syntax tree of the `[InterceptsLocation]` usage, similar to how paths in `#line` directives are normalized. Let this normalized path be called `normalizedInterceptorPath`. If exactly one syntax tree in the compilation has a normalized path which matches `normalizedInterceptorPath` by ordinal string comparison, that is the *referenced syntax tree*. Otherwise, an error occurs.
+#### Location encoding
 
-`#line` directives are not considered when determining the call referenced by an `[InterceptsLocation]` attribute. In other words, the file path, line and column numbers used in `[InterceptsLocation]` are expected to refer to *unmapped* source locations.
+The arguments to `[InterceptsLocation]` are:
+1. a version number. The compiler may introduce new encodings for the location in the future, with corresponding new version numbers.
```

Comment:
Was there a discussion of embedding the version number directly in the opaque data string? Maybe `<version>-<data>`? 
The version number doesn't seem especially useful when I look at an attribute like `[InterceptsLocation(1, "yPU5+1/pMuRHlz+XbnIQwQYAAAARAAAAUHJvZ3JhbS5jcw==")]`

Also, consider adding an example of location encoding in the doc to show what it looks like.

Reply:
It was thought that this gives the ability to change the precise encoding used in the data string, e.g. maybe in the future we want something different than base64, splitting out the version gives us the ability to express that.

Reply:
Looked again and I think I better understand the suggestion, which was to have a prefix that isn't necessarily encoded the same way. Yes, it is expressible either way.

The only reason I can think of to keep it this way is to separate parts that we think should be user-readable from those that are not, when it comes to diagnosing version incompatibility issues (which would arise from compiling later interceptors code in an earlier compiler.)